syntax = "proto3";

package analysis;

service AnalysisService {
    rpc AnalyzeText(TextRequest) returns (TextAnalysisResponse);
}

message TextRequest {
    string text = 1;
    repeated string archaic_words = 2;
}

message TextAnalysisResponse {
    float avg_word_length = 1;
    float avg_sentence_length = 2;
    map<string, int32> word_frequencies = 3;
}
